### ISSUE TITLE
zeroconf support

### ISSUE THREAD

**Issue by Yamakaky** [recalbox/recalbox-os#122]
Avahi could be installed to announce the raspi's ip address. Ssh to it would be simply `ssh recalbox.local`.

**Comment by digitalLumberjack** [comment]
Hi, I could also add many many things everyone could want but at the end it
will be as long to boot as a raspbian.
If one knows how to ssh he could be able to get the ip. And plus there is
an issue that will display the ip in network settings.

**Comment by Yamakaky** [comment]
Maybe we could start more things on demand ? On my server, ssh is started only when there is a connection on port 22.

**Comment by hstampfl** [comment]
You could also build a solution like this: https://learn.adafruit.com/downloads/pdf/drive-a-16x2-lcd-directly-with-a-raspberry-pi.pdf

**Comment by arthurlutz** [comment]
Am in favor of activating zero conf avahi-deamon so that one can easily find the recallbox on the network. 

I understand the will to have a lightweight system, but it would benefit hackers and debuggers to have some extra tooling that can be enabled in the config or via the configuration UI.

**Comment by nadenislamarre** [comment]
in 4.0.0 avahi is added for airplay,
i confirm that recalbox.local responds ;-)
the issue can be closed.

**Comment by Yamakaky** [comment]
Yeay !

**Comment by nwildner** [comment]
I know this may sound wrong, posting on an already closed issue, but i prefer posting here than opening another issue incorrectly.

What about using the `.lan` domain instead of `.local` domain?

Reason: First cause `.lan`  [is more apropriate top-level-domain](https://tools.ietf.org/html/rfc6762#appendix-G) reserved to internal networks. Second, some equipaments can have issues while using ".local" domains(i've faced some [macbooks having connectivity issues](
https://miketelahun.wordpress.com/2012/09/16/stop-using-local-as-the-top-level-domain-for-your-lan/) on a network i was called to take a look at these DNS problems)

> Quoting the RFC:
> 
>    Using ".local" as a private top-level domain conflicts with Multicast 
>    DNS and may cause problems for users. Clients can be configured 
>    to send both Multicast and
>    Unicast DNS queries in parallel for these names, and this does allow
>    names to be looked up both ways, but this results in additional
>    network traffic and additional delays in name resolution, as well as
>    potentially creating user confusion when it is not clear whether any
>    given result was received via link-local multicast from a peer on the
>    same link, or from the configured unicast name server.  Because of
>    this, we recommend against using ".local" as a private Unicast DNS
>    top-level domain.  We do not recommend use of unregistered top-level
>    domains at all, but should network operators decide to do this, the
>    following top-level domains have been used on private internal
>    networks without the problems caused by trying to reuse ".local." for
>    this purpose:
> 
>       .intranet.
>       .internal.
>       .private.
>       .corp.
>       .home.
>       .lan.
> 

What about `http://recalbox.lan`  :)

Other sources: https://serverfault.com/questions/17255/top-level-domain-domain-suffix-for-private-network - First comment about Apple clients having problem on resolving this.
